### PR TITLE
GitHub Actions

### PR DESCRIPTION
**Pull request by pjk25**
## Proposed Changes

Run the test suite via Actions on each push

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [x] Testing

## Further Comments

Workflow was generated using `make github-actions`, and remove the CHECK SCRIPTS step that would invoke `make bats`.

### CODE REVIEW

File: .github/workflows/test-erlang-otp-21.3.yaml [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+          path: deps
+          key: otp-21.3_git-${{ github.sha }}_deps
+      - name: FORMAT GIT REF
+        # https://github.community/t5/GitHub-Actions/How-can-I-set-an-expression-as-an-environment-variable-at/m-p/41804/highlight/true#M4751
+        id: ref
+        run: |
+          branch_or_tag_name=${GITHUB_REF#refs/*/}
+          echo "::set-output name=branch_or_tag_name::$branch_or_tag_name"
+      - name: UPLOAD TO S3
+        if: github.ref == 'refs/heads/master'
```
Comment (gerhard) [inline .github/workflows/test-erlang-otp-21.3.yaml:538]: Let's comment this (for both workflows) and double-check that it works before we uncomment and merge.

**Comment by gerhard** [comment]
Apart from the missing secrets required for the S3 upload (I've just added them now), it all looks good to me. I've left an inline comment re this.

Does this look good to you @dumbbell ?

**Comment by gerhard** [comment]
Oh, and let's add some badges via shields.io, e.g. https://github.com/rabbitmq/rabbitmq-prometheus/commit/d0378011a986bdd4077000b3a8d141b30ca6c41b

**Comment by pjk25** [comment]
@gerhard There is some flakiness to the rabbit_mgmt_test_db suite. Before we add the badges I'd like to see if I can fix that.

**Comment by pjk25** [comment]
@gerhard I've encountered the test flake locally when running the whole suite, so there is something worth fixing, although maybe not as part of this PR.

As for removing Travis, I think we should go ahead since we have equivalent coverage with Actions. It's not as if running both improves our confidence - if anything it could add confusion, and it's slower.

And let's use the GitHub badges rather than have a workaround. It's easy to change the name, but then we may hit that issue again later.

**Comment by pjk25** [comment]
They failure on Travis is reproducible locally when you run the whole suite. Afterwards, if I `ps aux | grep erl`, it looks like a test isn't cleaning up correctly:
```
paralle+ 16435  1.2  0.5 3244044 89984 ?       Sl   09:37   0:08 ct_run -W w -K true -A 64 -MBas ageffcbf -MHas ageffcbf -MBlmbcs 512 -MHlmbcs 512 -MMmcs 30 -P 1048576 -t 5000000 -stbt db -zdbbl 128000 -S 2 -sbwt very_short -A 24 -B i -- -root /usr/lib/erlang -progname erl -- -home /home/parallels/workspace/github.com/rabbitmq/rabbitmq-public-umbrella/deps/rabbitmq_management/logs/ct_run.ct_rabbitmq_management@parallels-Parallels-Virtual-Platform.2020-05-01_09.24.21/deps.rabbitmq_management.rabbit_mgmt_rabbitmqadmin_SUITE.logs/run.2020-05-01_09.37.08/log_private/ -- -pa  -noshell -noinput -s rabbit boot -boot start_sasl -ra wal_sync_method sync -lager crash_log false -lager handlers [] --
paralle+ 16524  0.0  0.0   4532  1512 ?        Ss   09:37   0:00 erl_child_setup 1024
paralle+ 19593  0.0  0.0  21536  1004 pts/0    S+   09:49   0:00 grep --color=auto erl
```

Which I think tripped me up on a different day, so I will see if I can fix it.

**Comment by gerhard** [comment]
> @gerhard I've encountered the test flake locally when running the whole suite, so there is something worth fixing, although maybe not as part of this PR.

👍🏻
 
> And let's use the GitHub badges rather than have a workaround. It's easy to change the name, but then we may hit that issue again later.

The lack of proper alignment is the reason why GitHub Actions badges are a turn-off for me:

image

This is what we (+@dumbbell) did for rabbitmq-server https://github.com/rabbitmq/rabbitmq-server/commit/cb99072d7e9f6ffa5bbfe0be78a1cf5c587942a3, and this is the end-result:

image

Notice that we have changed the workflow name to e.g. `Test - Erlang 21.3`

**Comment by pjk25** [comment]
Alignment was actually fine in the IntelliJ preview, which is interesting. I'll switch it back over and rename the workflow.

**Comment by pjk25** [comment]
@dumbbell fixed the flakiness with https://github.com/rabbitmq/rabbitmq-management/commit/9100530f9884239b2edf93277a16934c3174f573, and I have cleaned up the branch history, so I think this is now good to go @gerhard

**Comment by michaelklishin** [comment]
@gerhard should we merge this?

**Comment by gerhard** [comment]
@michaelklishin a merge is definitely long overdue. As soon as the checks that are currently running pass, I will merge.

**Pull request merged**